lib/riiif/routes.rb

```diff
@@ -17,21 +17,36 @@ def draw
         resource = options.fetch(:resource)
         route_prefix = options[:at]
         route_prefix ||= "/#{options[:as]}" if options[:as]
-        get "#{route_prefix}/:id/:region/:size/:rotation/:quality.:format" => 'riiif/images#show',
-            constraints: { rotation: ALLOW_DOTS, size: SIZES },
-            defaults: { format: 'jpg', rotation: '0', region: 'full', quality: 'default', model: resource },
-            as: options[:as] || 'image'
 
-        get "#{route_prefix}/:id/info.json" => 'riiif/images#info',
-            defaults: { format: 'json', model: resource },
-            as: [options[:as], 'info'].compact.join('_')
+        if route_prefix && route_prefix.starts_with?('http')
+          direct options[:as] || 'image' do |opts|
+            URI.join(route_prefix, ::File.join(opts[:id], opts[:region] || 'full', opts[:size], opts[:rotation] || '0', "#{opts[:quality] || 'default'}.#{opts[:format] || 'jpg'}")).to_s
```

Comment:
Something's wonky here.  With these settings:
```ruby
  Riiif::Engine.config.iiif_routes = { at: ENV["IIIF_BASE_URL"] }

  Riiif::Image.file_resolver = Class.new do
    def initialize(*_args); end

    def find(*_args); end
  end.new

  Riiif::Image.info_service = lambda do |id, _image|
    Riiif::Image.cache.fetch(
      Riiif::Image.cache_key(id, info: true),
      compress: true,
      expires_in: Riiif::Image.expires_in
    ) do
      uri = URI("#{ENV["IIIF_BASE_URL"]}/#{id}/info.json")
      JSON.parse(Net::HTTP.get_response(uri).body)
    end
  end
```
The `opts` variable is an integer referring to the image ID, not a hash at all.

Reply:
I think this might be the correct syntax:
```ruby 
          direct (options[:as] || 'image') do |id, params|
            URI.join(
              route_prefix,
              ::File.join(
                id.to_s,
                params[:region] || 'full',
                params[:size],
                params[:rotation] || '0',
                "#{params[:quality] || 'default'}.#{params[:format] || 'jpg'}"
              )
            ).to_s
          end

          direct [options[:as], 'info'].compact.join('_') do |id|
            URI.join(route_prefix, ::File.join(id.to_s, 'info.json')).to_s
          end

          direct [options[:as], 'base'].compact.join('_') do |id|
            URI.join(route_prefix, id.to_s).to_s
          end
```

But now there's a further problem.  My application specifies `mount Riiif::Engine => "/images", as: "riiif"` but with this branch and the changes above, image URLs are constructed as `/iiif/:id/etc` instead of `/images/:id`.  I have no idea what about this PR could be causing that change.